Description: Fix sediffx crash
Bug-Redhat: https://bugzilla.redhat.com/show_bug.cgi?id=832660
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/setools/+bug/1220618
Forwarded: no

--- a/sediff/progress.c
+++ b/sediff/progress.c
@@ -99,7 +99,7 @@ void progress_hide(progress_t * progress
 
 int progress_wait(progress_t * progress)
 {
-	GTimeVal wait_time = { 0, 50000 };
+	gint64 end_time;
 	g_mutex_lock(progress->mutex);
 	while (!progress->done) {
 		if (progress->s != NULL) {
@@ -111,7 +111,8 @@ int progress_wait(progress_t * progress)
 		// still be loading, and this dialog should not block
 		// until the entire source has been read
 		gtk_main_iteration_do(FALSE);
-		g_cond_timed_wait(progress->cond, progress->mutex, &wait_time);
+		end_time = g_get_monotonic_time () + 50000; // need to be set before each wait
+		g_cond_wait_until(progress->cond, progress->mutex, end_time);
 	}
 	g_mutex_unlock(progress->mutex);
 	if (progress->done < 0) {
